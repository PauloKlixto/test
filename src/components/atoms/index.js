import BaseBackButton from './BaseBackButton/BaseBackButton'
import BaseButton from './BaseButton/BaseButton'
import BaseCreditCard from './BaseCreditCard/BaseCreditCard'
import BaseHeading from './BaseHeading/BaseHeading'
import BaseSelect from './BaseSelect/BaseSelect'
import BaseSteps from './BaseSteps/BaseSteps'

export {
  BaseBackButton,
  BaseButton,
  BaseCreditCard,
  BaseHeading,
  BaseSelect,
  BaseSteps
}
