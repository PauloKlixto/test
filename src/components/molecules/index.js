import InputGroup from './InputGroup/InputGroup'
import DynamicForm from './DynamicForm/DynamicForm'

export { InputGroup, DynamicForm }
