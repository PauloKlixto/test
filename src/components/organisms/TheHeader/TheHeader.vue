<template>
  <section class="TheHeader">
    <section class="TheHeader__container">
      <Icon name="logo" />
      <ul class="TheHeader__mockMenu">
        <li class="TheHeader__mockMenuItem" v-for="(item, i) in 5" :key="i" />
      </ul>
    </section>
  </section>
</template>

<script>
export default {
  name: 'TheHeader'
}
</script>

<style lang="scss" scoped>
.TheHeader {
  /* Box model */
  display: grid;
  grid-template-columns: 1fr minmax(80%, 1280px) 1fr;
  grid-template-areas: '. container .';

  /* Visual */
  background: $color-white;
  border-bottom: 1px solid $color-light;

  &__container {
    /* Box model */
    grid-area: container;
    display: grid;
    grid-template-columns: 1fr 560px;
    align-items: center;

    @media (max-width: $viewport-md) {
      grid-template-columns: 1fr;
    }
  }

  &__mockMenu {
    /* Box model */
    display: grid;
    grid-template-columns: repeat(5, 1fr);
    grid-gap: $space-md;
    padding: 0;
    margin: 0;

    /* Visual */
    list-style: none;

    @media (max-width: $viewport-md) {
      display: none;
    }
  }

  &__mockMenuItem {
    /* Box model */
    width: 100px;
    height: 18px;

    /* Visual */
    background: $color-light;
    border-radius: 5px;
  }
}
</style>
