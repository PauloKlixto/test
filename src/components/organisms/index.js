import TheHeader from './TheHeader/TheHeader'

export { TheHeader }
