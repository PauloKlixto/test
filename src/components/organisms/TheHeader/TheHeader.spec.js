/* eslint-disable no-undef */
import { shallowMount, createLocalVue } from '@vue/test-utils'
import TheHeader from './TheHeader'
import { Icon } from '@/components/photons'

const localVue = createLocalVue()
localVue.component('Icon', Icon)

describe('TheHeader', () => {
  const WRAPPER = shallowMount(TheHeader, { localVue })

  test('Componente existe', () => {
    expect(WRAPPER.exists()).toBe(true)
  })

  test('Componente nÃ£o alterado', () => {
    expect(WRAPPER.html()).toMatchSnapshot()
  })
})
