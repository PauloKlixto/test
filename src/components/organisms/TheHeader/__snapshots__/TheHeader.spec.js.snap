// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheHeader Componente n√£o alterado 1`] = `
"<section class=\\"TheHeader\\">
  <section class=\\"TheHeader__container\\">
    <icon-stub name=\\"logo\\" path=\\"icons\\"></icon-stub>
    <ul class=\\"TheHeader__mockMenu\\">
      <li class=\\"TheHeader__mockMenuItem\\"></li>
      <li class=\\"TheHeader__mockMenuItem\\"></li>
      <li class=\\"TheHeader__mockMenuItem\\"></li>
      <li class=\\"TheHeader__mockMenuItem\\"></li>
      <li class=\\"TheHeader__mockMenuItem\\"></li>
    </ul>
  </section>
</section>"
`;
